lib/imagemounter/utils/list_developer_image.js

```diff
@@ -0,0 +1,182 @@
+ */
+function findDeveloperImageFromDirectory(entry) {
+    const fileList = readdirSync(entry);
+    for (const subEntry of fileList) {
+        const fullPath = joinPath(entry, subEntry);
+        const statResult = lstatSync(fullPath);
+        if (subEntry === DEVELOPER_IMAGE_FILE_NAME && statResult.isFile()) {
+            return entry;
+        }
+        if (statResult.isDirectory()) {
+            const subFolderResult = findDeveloperImageFromDirectory(fullPath);
+            if (subFolderResult) {
+                return subFolderResult;
+            }
+        }
+    }
+    return undefined;
+}
+
+/**
+ * @typedef {Object} ImagePath
+ * @property {string} developerImage
+ * @property {string} developerImageSignature
+ */
+
+/**
+ * Find developer image for certain version. If developer image does not exists,
+ * this will try to find and download developer image, unzip to `~/.appium/iOSSupport/`
+ * @param {string} version full version of iOS device
+ * @returns {ImagePath}
+ * @throws If developer image is not found, or error while downloading or unzipping.
+ */
+async function findDeveloperImage(version) {
+    const finalVersion = version.split('.').splice(0, 2).join('.');
+    const fileName = `${finalVersion}.zip`;
+    const fullDownloadPath = joinPath(DEFAULT_IMAGE_DIR, fileName);
+    if (!existsSync(DEFAULT_IMAGE_DIR)) {
+        mkdirSync(DEFAULT_IMAGE_DIR, { recursive: true });
+    }
+    if (!existsSync(fullDownloadPath)) {
+        await searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath);
+    }
+    const decompressPath = joinPath(DEFAULT_IMAGE_DIR, finalVersion);
+    let developerImageParentFolder = null;
+    if (existsSync(decompressPath)) {
+        developerImageParentFolder = findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        await zip.extractAllTo(fullDownloadPath, decompressPath);
+        developerImageParentFolder = findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        throw Error(`Unable to find unzipped developer image in ${decompressPath}`);
+    }
+    return {
+        developerImage: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_FILE_NAME),
+        developerImageSignature: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_SIGNATURE_FILE_NAME)
+    };
+}
+
+async function searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath) {
+    const fileNameRegExp = new RegExp(`${finalVersion}(\\(([\\w_|.()])+\\))?.zip`);
+    let fileUrl = null;
+    for (const repo of AVAILABLE_REPOS) {
+        const fileList = await listGithubImageList(repo.user, repo.name, repo.subFolderList);
+        if (!fileList) {
+            continue;
+        }
+        const targetFile = _.find(fileList, (item) => fileNameRegExp.test(item.path));
+        if (targetFile != null) {
```

Comment:
always use strict comparison (`!==`)

Reply:
This should be simplified as `if (targetFile)`